### PR TITLE
Allow hostNetwork ingress for network policy

### PR DESCRIPTION
**Pull request by jboxman**
- https://bugzilla.redhat.com/show_bug.cgi?id=1768608

Based on https://github.com/openshift/openshift-docs/pull/18871.

cc @juanluisvaladas

### CODE REVIEW

**Comment by jboxman** [comment]
@juanluisvaladas, I updated the content. Can you take a look? Thanks!

**Comment by juanluisvaladas** [comment]
/lgtm
@jboxman looks good, and very explicit.

**Comment by jboxman** [comment]
@weliang1, can you verify this allows network policy to work with a host network Ingress Controller?

Thanks!

File: modules/nw-networkpolicy-multitenant-isolation.adoc [hunk 1]
```diff
@@ -68,6 +68,21 @@ $ oc apply -f <policy-name>.yaml \ <1>
 <2> Replace `<project>` with the name of the project to apply the NetworkPolicy
 object to.
 
+. If the `default` Ingress Controller configuration has the `spec.endpointPublishingStrategy: HostNetwork` value set, you must apply a label to the the `default` {product-title} namespace to allow network traffic between the Ingress Controller and the project:
+
+.. Determine if your `default` Ingress Controller uses the `HostNetwork` endpoint publishing strategy:
++
+----
+$ oc get --namespace openshift-ingress-operator ingresscontrollers/default \
```
Comment (weliang1) [inline modules/nw-networkpolicy-multitenant-isolation.adoc]: The correct oc command is:
[root@dhcp-41-193 FILE]# oc get --namespace openshift-ingress-operator ingresscontrollers/default --output jsonpath='{.status.endpointPublishingStrategy}'
map[loadBalancer:map[scope:External] type:LoadBalancerService]
[root@dhcp-41-193 FILE]# oc get --namespace openshift-ingress-operator ingresscontrollers/default --output jsonpath='{.status.endpointPublishingStrategy.type}'
LoadBalancerService

Comment (jboxman) [reply modules/nw-networkpolicy-multitenant-isolation.adoc]: Thanks @weliang1!

**Comment by jboxman** [comment]
@openshift/team-documentation, PTAL. Thanks!

**Comment by jboxman** [comment]
@weliang1, I made the suggested change; Can you confirm? Thanks!

File: modules/nw-networkpolicy-about.adoc [hunk 2]
```diff
@@ -68,11 +68,18 @@ spec:
   policyTypes:
   - Ingress
 ----
+
++
+If the Ingress Controller is configured with `endpointPublishingStrategy: HostNetwork`, then the Ingress Controller Pod runs on the host network.
+When running on the host network, the traffic from the Ingress Controller is assigned `netid:0`.
+The `netid` for the namespace associated with the Ingress Operator is different, so the `matchLabel` in the `allow-from-openshift-ingress` network policy will not work as expected.
```
Comment (kalexand-rh) [inline modules/nw-networkpolicy-about.adoc]: s/associated/that is associated

s/will not/does not

At this point, will the user know what is expected?

File: modules/nw-networkpolicy-about.adoc [hunk 3]
```diff
@@ -68,11 +68,18 @@ spec:
   policyTypes:
   - Ingress
 ----
+
++
+If the Ingress Controller is configured with `endpointPublishingStrategy: HostNetwork`, then the Ingress Controller Pod runs on the host network.
+When running on the host network, the traffic from the Ingress Controller is assigned `netid:0`.
```
Comment (kalexand-rh) [inline modules/nw-networkpolicy-about.adoc]: s/`netid:0`/ the `netid:0` _appropriate noun_

File: modules/nw-networkpolicy-about.adoc [hunk 4]
```diff
@@ -68,11 +68,18 @@ spec:
...
+The `netid` for the namespace associated with the Ingress Operator is different, so the `matchLabel` in the `allow-from-openshift-ingress` network policy will not work as expected.
+Because the `default` namespace is assigned `netid:0`, you can allow traffic from the `default` Ingress Controller by labeling your `default` namespace with `network.openshift.io/policy-group: ingress`.
```
Comment (kalexand-rh) [inline modules/nw-networkpolicy-about.adoc]: s/`netid:0`/ the `netid:0` _same appropriate noun_

**Review by kalexand-rh — COMMENTED** [review]
Couple of picks, then this looks good.

**Pull request merged**